centreon/src/Core/Dashboard/Playlist/Application/UseCase/CreatePlaylist/CreatePlaylist.php

```diff
@@ -0,0 +1,221 @@
+            $presenter->presentResponse(new ErrorResponse(PlaylistException::errorWhileCreating()));
+        }
+    }
+
+    /**
+     * @param Playlist $playlist
+     *
+     * @return CreatePlaylistResponse
+     */
+    private function createResponse(Playlist $playlist): CreatePlaylistResponse
+    {
+        $response = new CreatePlaylistResponse();
+
+        $response->id = $playlist->getId();
+        $response->name = $playlist->getName();
+        $response->description = $playlist->getDescription();
+        $response->dashboards = array_map(function (DashboardOrder $dashboardOrder) {
+            return [
+                'id' => $dashboardOrder->getDashboardId(),
+                'order' => $dashboardOrder->getOrder(),
+            ];
+        }, $playlist->getDashboardsOrder());
+        $response->rotationTime = $playlist->getRotationTime();
+        $response->isPublic = $playlist->isPublic();
+        if ($playlist->getAuthor() !== null) {
+            $response->author = [
+                'id' => $playlist->getAuthor()->getId(),
+                'name' => $playlist->getAuthor()->getName(),
+            ];
+        }
+        $response->createdAt = $playlist->getCreatedAt();
+
+        return $response;
+    }
+
+    /**
+     * Validate that name is unique and dashboards exists.
+     *
+     * @param CreatePlaylistRequest $request
+     *
+     * @throws PlaylistException|\Throwable
+     */
+    private function validateNameAndDashboards(CreatePlaylistRequest $request): void
+    {
+        $this->info(
+            'Validating playlist name and dashboards',
+            [
+                'playlist_name' => $request->name,
+                'dashboards' => $request->dashboards,
+            ]
+        );
+        $this->validator->validatePlaylistNameIsUnique($request->name);
+        if ([] !== $request->dashboards) {
+            $dashboardIds = array_map(static fn (array $dashboard): int => $dashboard['id'], $request->dashboards);
+            $this->validator->validateDashboardExists($dashboardIds);
+            $this->validator->validateDashboardIsUnique($dashboardIds);
+            if (! $this->rights->hasAdminRole()) {
+                $this->validator->validateUserHasAccessToDashboards($dashboardIds, $this->user);
+            }
+        }
+    }
+
+    /**
+     * Create NewPlaylist Entity.
+     *
+     * @param CreatePlaylistRequest $request
+     *
+     * @throws \Assert\AssertionFailedException
+     * @throws PlaylistException
+     *
+     * @return NewPlaylist
+     */
+    private function createNewPlaylistModel(CreatePlaylistRequest $request): NewPlaylist
+    {
+        $newPlaylist = (new NewPlaylist($request->name, $request->rotationTime, $request->isPublic))->setAuthor(
+            new PlaylistAuthor($this->user->getId(), $this->user->getAlias())
+        );
+        if ($request->description !== null) {
+            $newPlaylist->setDescription($request->description);
+        }
```

Comment:
Useless if because the method accepts `null`
```suggestion
        $newPlaylist->setDescription($request->description);
```